File: backend/src/main/java/com/reto/backend/security/dto/JwtDTO.java
package com.reto.backend.security.dto;

import lombok.Data;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;

@Data
public class JwtDTO {
    private String token;
    private String bearer = "Bearer";
    private String username;
    private Collection<? extends GrantedAuthority> authorities;

    public JwtDTO(String token, String username, Collection<? extends GrantedAuthority> authorities) {
        this.token = token;
        this.username = username;
        this.authorities = authorities;
    }
}

File: front/src/app/models/JwtDTO.ts
export class JwtDTO {
    token: string;
    type: string;
    username: string;
    authorities: string[];

    constructor(token: string,type: string,username: string,authorities: string[]){
        this.token = token;
        this.type = type;
        this.username = username;
        this.authorities = authorities;
    }
}